sample/sample_test.go

```diff
@@ -0,0 +1,184 @@
+package sample
+
+import (
+	"fmt"
+	"math"
+	"slices"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestTemperature(t *testing.T) {
+	logits, err := Temperature(0.5).Apply([]float64{-3, -2, -1, 0, 1, 2, 4})
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := []float64{-14, -12, -10, -8, -6, -4, 0}
+	if !floats.Equal(logits, want) {
+		t.Fatalf("got: %v, want: %v", logits, want)
+	}
+
+	if _, err := Temperature(-1).Apply([]float64{-3, -2, -1, 0, 1, 2, 4}); err == nil {
+		t.Fatalf("expected error for temperature=-1, got %v", logits)
+	}
+	if _, err := Temperature(2.1).Apply([]float64{-3, -2, -1, 0, 1, 2, 4}); err == nil {
+		t.Fatalf("expected error for temperature=2.1, got %v", logits)
+	}
+}
+
+func TestSoftmax(t *testing.T) {
+	probs := softmax([]float64{-3, -2, -1, 0, 1, 2, 4})
+
+	expectedProbs := []float64{0.000751406628089903, 0.0020425349829204676, 0.005552185728064613, 0.015092405572827691, 0.04102541181635154, 0.11151863144543739, 0.8240174238263085}
+	if !floats.Equal(probs, expectedProbs) {
+		t.Fatalf("logits: %v, expectedlogits: %v", probs, expectedProbs)
+	}
+}
+
+func TestTopK(t *testing.T) {
+	logits, err := TopK(3).Apply([]float64{-3, -2, -1, 0, 1, 2, 4})
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedlogits := []float64{math.Inf(-1), math.Inf(-1), math.Inf(-1), math.Inf(-1), 1, 2, 4}
+	if !floats.Same(logits, expectedlogits) {
```

Comment:
Use `slices.Equal` though I personally prefer `cmp.Diff` since you can output the difference